src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKModelineInsertionParser.java

```diff
@@ -0,0 +1,62 @@
+package com.github.cameltooling.lsp.internal.parser;
+
+import com.github.cameltooling.lsp.internal.completion.modeline.CamelKModelineInsertionProcessor;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class CamelKModelineInsertionParser {
+
+    public boolean canPutCamelKModeline(Position position, TextDocumentItem textDocumentItem) {
+        int currentLine = position.getLine();
+
+        //Save the lines to not have to access multiple times to file?
```

Comment:
Performance comes later. We have no performance issue identified yet. No need to struggle with it too much.There are tons of places where it can be improved. Without a proper performance analysis we do not know if it is an important place to work on or not.
If you have suspicions of an important impact, please report a specific issue to do a performance analysis on it. (but I do not think that it will have an high priority soon)

Also this comment can become obsolete after a refactor without reading this piece of code. Making the comment outdated and misleading.